### PR TITLE
refactor(core): allow developers to select static-query migration strategy

### PR DESCRIPTION
**Pull request by devversion**
* See individual commits

### CODE REVIEW

File: packages/core/schematics/migrations/static-queries/index.ts [hunk 1]
```diff
@@ -22,34 +23,82 @@ import {TimingStrategy} from './strategies/timing-strategy';
...
+
+  // In case prompts are supported, determine the desired migration strategy
+  // by creating a choice prompt. By default the template strategy is used.
+  let isUsageStrategy = false;
+  if (supportsPrompt()) {
+    logger.info('There are two available migration strategies that can be selected:');
+    logger.info('  • Template strategy  -  (safe migration, uses anti-patterns)');
+    logger.info('  • Usage strategy  -  (uses best practices, not always correct)');
+    logger.info('For an easy migration, the template strategy is recommended. The usage');
+    logger.info('strategy can be used for best practices and easier migration to Ivy.');
```
Comment (kara) [inline packages/core/schematics/migrations/static-queries/index.ts]: Template strategy - migration tool (short-term gains, rare corrections)
Usage strategy - best practices (long-term gains, manual corrections)

```suggestion
    logger.info('strategy can be used for best practices and a code base that will be more flexible to changes going forward.');
```

**Review by kara — APPROVED** [review]
LGTM once message is fixed

**Review by alexeagle — APPROVED** [review]
looks like a clever way to make the prompts work from core

File: packages/core/schematics/migrations/static-queries/index.ts (hunk 1 above)
Comment (devversion) [reply packages/core/schematics/migrations/static-queries/index.ts]: Sounds good to me. done!

**Comment by benlesh** [comment]
[presubmit](http://test/OCL:244095195:BASE:244095212:1555545630802:74cc8602)

**Comment by benlesh** [comment]
Issues syncing this in google3, will discuss with other googlers.

**Comment by benlesh** [comment]
Was able to fix google3 downstream. Good to merge.

**Pull request closed without merging**